test/acceptance/user.controller.acceptance.ts

```diff
@@ -130,6 +133,74 @@ describe('UserController', () => {
     });
   });
 
+  describe('authentication functions', () => {
+    it('login returns a valid token', async () => {
+      const newUser = await userRepo.create(user);
+      await client
+        .post('/users/login')
+        .send({email: newUser.email, password: newUser.password})
+        .expect(200)
+        .then(getToken);
+
+      function getToken(res: Response) {
+        // is it good to return the token as text?
+        const token = res.text;
+        expect(token).to.not.be.empty();
+      }
+    });
+
+    it('login returns an error when invalid credentials are used', async () => {
+      const newUser = await userRepo.create(user);
+      newUser.password = 'wrong password';
+      await client
+        .post('/users/login')
+        .send({email: newUser.email, password: newUser.password})
+        .expect(401);
+    });
+
+    it('/me returns the current user', async () => {
+      const newUser = await userRepo.create(user);
+      let token = '';
+      await client
+        .post('/users/login')
+        .send({email: newUser.email, password: newUser.password})
+        .expect(200)
+        .then(getToken);
+
+      function getToken(res: Response) {
+        // is it good to return the token as text?
+        token = res.text;
+      }
+      // MongoDB returns an id object we need to convert to string
+      // since the REST API returns a string for the id property.
+      newUser.id = newUser.id.toString();
+      const me = _.pick(newUser.toJSON(), ['id', 'email']);
+
+      await client
+        .get('/users/me')
+        .set('Authorization', token)
+        .expect(200, me);
+    });
+
+    it('/me returns 401 when the token is not provided', async () => {
+      const newUser = await userRepo.create(user);
+      // tslint:disable-next-line:no-unused-variable
+      let token = '';
+      await client
+        .post('/users/login')
+        .send({email: newUser.email, password: newUser.password})
+        .expect(200)
+        .then(getToken);
```

Comment:
Use `await` instead of `then`.

```ts
const res = await client
    .post('/users/login')
    .send({email: newUser.email, password: newUser.password})
    .expect(200);
const token = res.text;
```

Even better, rewrite this code to create the token by calling UserRepository API directly, no need to make HTTP calls here.